src/object/authenticate_access_auth.cpp

```diff
@@ -1136,12 +1175,691 @@ au_user_revoke_all_privileges (MOP user_mop)
+    {
+    case DB_OBJECT_CLASS:
+      sprintf (obj_fetch_query, sql_query, "SELECT [cl].[class_of] FROM " CT_CLASS_NAME "[cl] WHERE [unique_name] = ?");
+      break;
+    case DB_OBJECT_PROCEDURE:
+      sprintf (obj_fetch_query, sql_query, "SELECT [sp] FROM " CT_STORED_PROC_NAME "[sp] WHERE [unique_name] = ?");
+      break;
+    default:
+      error = ER_FAILED;
+      ASSERT_ERROR_AND_SET (error);
+      goto exit;
+    }
+
+  session = db_open_buffer_local (obj_fetch_query);
+  if (session == NULL)
+    {
+      ASSERT_ERROR_AND_SET (error);
+      goto exit;
+    }
+
+  error = db_set_system_generated_statement (session);
+  if (error != NO_ERROR)
+    {
+      goto release;
+    }
+
+  stmt_id = db_compile_statement_local (session);
+  if (stmt_id < 0)
+    {
+      ASSERT_ERROR_AND_SET (error);
+      goto release;
+    }
+
+  /* Prepare DB_VALUEs for host variables */
+  db_make_string (&val, unique_name);
+
+  error = db_push_values (session, 1, &val);
+  if (error != NO_ERROR)
+    {
+      ASSERT_ERROR_AND_SET (error);
+      goto release;
+    }
+
+  error = db_execute_statement_local (session, stmt_id, &result);
+
+  /* The error value is row count if it's not negative value. */
+  if (error == 0)
+    {
+      *row_count = error;
+      goto release;
+    }
+  else if (error < 0)
+    {
+      ASSERT_ERROR_AND_SET (error);
+      goto release;
+    }
+
+  *row_count = error;
+  error = NO_ERROR;
+
+  while (db_query_next_tuple (result) == DB_CURSOR_SUCCESS)
+    {
+      if (db_query_get_tuple_value (result, 0, &grantee_value) == NO_ERROR)
+	{
+	  if (!DB_IS_NULL (&grantee_value))
+	    {
+	      grantee_mop = db_get_object (&grantee_value);
+	    }
+	  else
+	    {
+	      goto release;
+	    }
+	}
+
+      if (db_query_get_tuple_value (result, 1, &object_of_value) == NO_ERROR)
+	{
+	  if (!DB_IS_NULL (&object_of_value))
+	    {
+	      object_of_mop = db_get_object (&object_of_value);
+	    }
+	  else
+	    {
+	      goto release;
+	    }
+	}
+
+      assert (grantee_mop != NULL && object_of_mop != NULL);
+
+      if (au_get_object (grantee_mop, "authorization", &auth) != NO_ERROR)
+	{
+	  error = ER_AU_ACCESS_ERROR;
+	  er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, error, 2, AU_USER_CLASS_NAME, "authorization");
+	  goto release;
+	}
+      else if (au_fetch_instance (auth, NULL, AU_FETCH_UPDATE, LC_FETCH_MVCC_VERSION, AU_UPDATE) != NO_ERROR)
+	{
+	  error = ER_AU_CANT_UPDATE;
+	  er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, error, 0);
+	  goto release;
+	}
+      else if ((error = obj_inst_lock (auth, 1)) == NO_ERROR && (error = get_grants (auth, &grants, 1)) == NO_ERROR)
+	{
+	  gsize = set_size (grants);
+	  for (gindex = 0; gindex < gsize && error == NO_ERROR; gindex += GRANT_ENTRY_LENGTH)
+	    {
+	      error = set_get_element (grants, GRANT_ENTRY_CLASS (gindex), &element);
+	      if (error != NO_ERROR)
+		{
+		  ASSERT_ERROR_AND_SET (error);
+		  goto release;
+		}
+
+	      if (db_get_object (&element) == object_of_mop)
```

Comment:
```suggestion
	      if (ws_is_same_object (db_get_object (&element), object_of_mop))
```
다음과 같이 수정하여, 1413번째 if문 내의 작성하신 코드와 통일성을 주는 건 어떨까요?

Reply:
코드 리뷰 감사합니다.
리뷰 내용을 반영하여 수정했습니다.